.github/workflows/build_ts_client.yml

```diff
@@ -0,0 +1,83 @@
+name: Build and publish TypeScript API client
+
+on:
+  pull_request:
+  release:
+    types:
+      - published
+      - created
+      - edited
+      - prereleased
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      # Setup .npmrc file to publish to npm
+      - uses: actions/setup-node@v1
+        with:
+          node-version: '12.x'
+          registry-url: 'https://npm.pkg.github.com'
+      - name: Prepare tools for building TypeScript client
+        working-directory: client/typescript-axios
+        run: |
+          yarn install --frozen-lockfile
+      - name: Generate TypeScript API client
+        working-directory: client/typescript-axios
+        run: |
+          yarn openapi-generator generate -i ../../api.swagger.yaml -g typescript-axios -o build/src/ -c generator.yml
+      - name: Generate PR build version
+        if: github.event_name == 'pull_request'
+        working-directory: client/typescript-axios
+        run: |
+          echo ::set-env name=NEW_CLIENT_VERSION::`node pr-build-version.js ${{ github.base_ref }} pr-build-${{ github.run_id }}`
+      - name: Generate release build version
+        if: github.event_name == 'release'
+        working-directory: client/typescript-axios
+        run: |
+          echo ::set-env name=NEW_CLIENT_VERSION::`node pr-build-version.js ${{ github.ref }}`
```

Comment:
Are you sure `${{ github.ref }}` is a plain string? It seems to me it can come from a git branch name, and those allow a surprising range of characters...

Reply:
Ah, okay now I see what you mean.
According to [man page](https://mirrors.edge.kernel.org/pub/software/scm/git/docs/git-check-ref-format.html) whitespaces and special characters aren't allowed; what could be a problem are quote symbols (`'`, `"` or backquote) but:
a) I'm not sure if they could be easily escaped;
b) once again, the most horrible thing which could happen is build failure;
c) what a horrible person would want to name a branch like that?

How about adding quotes - do you think it would help (at least in some situations)?
```suggestion
          echo ::set-env name=NEW_CLIENT_VERSION::`node pr-build-version.js "${{ github.ref }}"`
```

Reply:
Can it be put in an env variable and referenced from there? If so then quotes should indeed help - otherwise they only really help with spaces, which as you correctly pointed out aren't allowed anyway.

Reply:
Here we go: 5be2cd0.